archunit/build.gradle

```diff
@@ -2,6 +2,9 @@ apply plugin: 'idea'
 
 ext.moduleName = 'com.tngtech.archunit'
 
+sourceCompatibility = JavaVersion.VERSION_1_7
```

Comment:
why is this necessary, by `archunit/build.gradle > subprojects {...}` this should be the default for all modules that do not explicitly override it, isn't it?

Reply:
Maybe your problem would also be gone without this now that you added the tests to `jdk9test`?